pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go

```diff
@@ -75,13 +75,61 @@ func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfig
 		return false, fmt.Errorf("cast of cp.Value failed: %v", okCpCast)
 	}
 
+	merged = false
+
+	// First merge transport socket matches with the same name
+	if (len(c.GetTransportSocketMatches())) > 0 && (len(cpValueCast.GetTransportSocketMatches()) > 0) {
+		for _, cpTsm := range cpValueCast.GetTransportSocketMatches() {
+			if cpTsm.GetName() == "" || (cpTsm.GetTransportSocket() == nil && cpTsm.GetMatch() == nil) {
+				continue
+			}
+
+			for _, tsm := range c.GetTransportSocketMatches() {
+				if tsm.GetName() == cpTsm.GetName() {
+					// Replace match if defined in patch
```

Comment:
I am not sure whether it is right

Reply:
Is this about the condition or just about replacing whole match?